_pytest/main.py

```diff
@@ -468,6 +475,11 @@ def _parsearg(self, arg):
         if self.config.option.pyargs:
             parts[0] = self._tryconvertpyarg(parts[0])
         relpath = parts[0].replace("/", os.sep)
+        pattern_line = r':(?P<line>[0-9]+)$'
```

Comment:
Please compile these regexes and store them into the function as attributes:

```python
def _find_nearest_test_function(path_expr):
    ...

_find_nearest_test_function.PATTERN_LINE = re.compile(r':(?P<line>[0-9]+)$')
```

This will make the calls slightly faster. This advice also applies to the regexes in `_get_nearest_func_name`.

Reply:
Fixed